Automattic/wp-calypso
# Pattern assembler: Refactor submit logic into separate action

Part of https://github.com/Automattic/dotcom-forge/issues/1912

## Proposed Changes

Moves the logic that applies a design created in the pattern assembler to a separate action so it can be re-used when selecting a pattern-based theme (this will be required by virtual themes i2, as part of https://github.com/Automattic/dotcom-forge/issues/1912).

## Testing Instructions

- Use the Calypso live link below
- Go to `/start`
- Create a new site with a Free plan
- Select the "Write & publish" goal
- Proceed in the boarding until you see an option to "View designs"
- Click on it to access the Design picker
- Scroll to the bottom and click on the Pattern Assembler CTA
- Create your own design
- Once you're done, click on the Continue button
- Make sure the design has been applied successfully (it should match the same behavior as production)
